### PR TITLE
Fixed error happening when a renaming is performed

### PR DESCRIPTION
**Pull request by MMaiero**
Changed the equality check to deal with case.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

### CODE REVIEW

**Comment by MMaiero** [comment]
Closes #1435

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java [hunk 1]
```diff
@@ -529,21 +529,21 @@ private void removeDeletedFromWireGraphProperty(final String pid) throws GwtKura
...
+                logger.info("Deleting Wire Component: PID -> {}", componentPid);
                 configService.deleteFactoryConfiguration(componentPid, false);
                 continue;
             }
             boolean isFound = false;
             for (int i = 0; i < length; i++) {
                 final JsonObject jsonObject = jWireGraph.get(String.valueOf(i)).asObject();
                 final String component = jsonObject.getString("pid", null);
-                if (component.equalsIgnoreCase(componentPid)) {
+                if (component.equals(componentPid)) {
```
Comment (dwoodard1) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:540]: @MMaiero I am fine merging this, but why was this originally equalsIgnoreCase? I just want to make sure we aren't breaking something more fundamental to fix this edge condition.

Comment (MMaiero) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:540]: I believe I've not contributed it. So maybe we can ask @amitjoy to give some hints

Comment (amitjoy) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:540]: Previously when I implemented this, I have added a constraint that names of wire components will be case-insesnsitive, that is, a wire component with a name `MyWC` would be as same as `mywc`. Therefore, no two components could co-exist with such names. Later on, I removed this constraint but its regression stayed. Now `equals` is more appropriate. Also note that, there are few more places where I used `equalsIgnoreCase` in a similar way. Make sure those also do not introduce any regression.

Comment (dwoodard1) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java:540]: Thanks Amit! @MMaiero Do you want to check for more instances of equalsIgnoreCase?

**Pull request merged**